contracts/contracts/VerificationGateway.sol

```diff
@@ -168,8 +168,7 @@ contract VerificationGateway
         IWallet wallet = IWallet(msg.sender);
         bytes32 existingHash = hashFromWallet[wallet];
         if (existingHash == bytes32(0)) { // wallet does not yet have a bls key registered with this gateway
-            // set it instantly
-            safeSetWallet(messageSenderSignature, publicKey, wallet, signatureExpiryTimestamp);
+            // Can't register new wallet contracts, only what this gateway deployed.
```

Comment:
This entire logically block would be easier to follow as a single if statement, since the original branch is no a no-op.
(couldn't use suggest changes)
```solidity
...
// Can't register new wallet contracts, only what this gateway deployed.
if (existingHash != bytes32(0)) { // wallet already has a key registered, set after delay
    pendingMessageSenderSignatureFromHash[existingHash] = messageSenderSignature;
    pendingBLSPublicKeyFromHash[existingHash] = publicKey;
    pendingBLSPublicKeyTimeFromHash[existingHash] = block.timestamp + 604800; // 1 week from now
    emit PendingBLSKeySet(existingHash, publicKey);
}
...
```